risc0/zkvm/src/exec/mod.rs

```diff
@@ -160,6 +179,18 @@ impl<'a> Executor<'a> {
 
     /// Run the executor until [ExitCode::Paused] or [ExitCode::Halted] is
     /// reached, producing a [Session] as a result.
+    /// # Example
+    /// ```
+    /// # use risc0_zkvm::{serde::to_vec, Executor, ExecutorEnv, Session};
+    /// # use risc0_zkvm_methods::{BENCH_ELF, bench::{BenchmarkSpec, SpecWithIters}};
+    /// #
+    /// # let spec = SpecWithIters(BenchmarkSpec::SimpleLoop, 1);
+    /// # let env = ExecutorEnv::builder()
+    /// #   .add_input(&to_vec(&spec).unwrap())
+    /// #   .build();
+    /// # let mut exec = Executor::from_elf(env, BENCH_ELF).unwrap();
```

Comment:
I think showing the `env` & `exec` gives useful context. If the `spec` and `BENCH_ELF` names seem confusing (and they probably are), this could be further upgraded by hidden lines like `let EXAMPLE_ELF = BENCH_ELF` and naming `spec` to instead by `example_input` or something similar.